### PR TITLE
add engine-clients module back into the runtime

### LINKED ISSUE #941: functional regression: grafana annotator client is missing
The --docker-metrics option auto-wires an http client to publish annotations to the configured grafana instance.
This was removed as part of the clients module, but this particular client needs to be restored.

### PR DESCRIPTION
**Pull request by jshook**
This PR makes the grafana annotator available in the runtime again.
closes #941

### CODE REVIEW

File: engine-clients/pom.xml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        <dependency>
+            <groupId>io.nosqlbench</groupId>
+            <artifactId>nb-api</artifactId>
+            <version>5.17.1-SNAPSHOT</version>
+            <scope>compile</scope>
+        </dependency>
+
+    </dependencies>
+
+    <build>
```
Comment (jeffbanks) [inline engine-clients/pom.xml:35]: Do want to include the jacoco here as well?

File: engine-clients/src/main/java/io/nosqlbench/engine/clients/grafana/analyzer/GrafanaRegionAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+public class GrafanaRegionAnalyzer implements Runnable {
+
+    private String baseUrl = "";
+    //    private String dashboardId = "";
+    private GrafanaClient gclient;
+
+    public static void main(String[] args) {
+        GrafanaRegionAnalyzer analyzer = new GrafanaRegionAnalyzer();
+        analyzer.setBaseUrl("http://44.242.139.57:3000/");
+//
```
Comment (jeffbanks) [inline engine-clients/src/main/java/io/nosqlbench/engine/clients/grafana/analyzer/GrafanaRegionAnalyzer.java:46]: Do we still want this entry?

File: engine-clients/src/main/java/io/nosqlbench/engine/clients/grafana/analyzer/GrafanaRegionAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            long id = mainpanel.getId();
+            String title = mainpanel.getTitle();
+            String description = mainpanel.getDescription();
+
+            List<GPanelDef> panels = mainpanel.getPanels();
+            String datasource = mainpanel.getDatasource();
+            Map<String, Object> fieldConfig = mainpanel.getFieldConfig();
+            Map<String, String> options = mainpanel.getOptions();
+            List<GPanelDef.GTarget> targets = mainpanel.getTargets();
+            System.out.println("targets:\n" + targets);
```
Comment (jeffbanks) [inline engine-clients/src/main/java/io/nosqlbench/engine/clients/grafana/analyzer/GrafanaRegionAnalyzer.java:132]: Do we want the Logger usage here?
There are quite a few system.out entries in this file.

File: engine-clients/src/test/java/io/nosqlbench/engine/clients/grafana/GrafanaClientTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    @Test
+    @Disabled
+    public void testCreateAnnotation() {
+        GrafanaClient client = new GrafanaClient(testurl);
+        client.getConfig().basicAuth("admin", "admin");
+        GAnnotation a = new GAnnotation();
+        a.setDashboardId(2);
+        a.setText("testingAnnotation");
+        GAnnotation created = client.createAnnotation(a);
+        System.out.println(created);
```
Comment (jeffbanks) [inline engine-clients/src/test/java/io/nosqlbench/engine/clients/grafana/GrafanaClientTest.java:37]: Logging usage can be applied in this file.

**Review by jeffbanks — APPROVED** [review]
Since this was a pre-existing module I'm not as concerned with the specific logic (Graphana) pulled back in.

Some items for consideration:
* Logging usage where `System.out` occurs.
* There's a bit of dead code existing.
* The copyright headers can be upgraded to 2022-2023.

File: engine-clients/pom.xml (hunk 1 above)
Comment (jshook) [reply engine-clients/pom.xml:35]: Yes, that would be good

File: engine-clients/src/main/java/io/nosqlbench/engine/clients/grafana/analyzer/GrafanaRegionAnalyzer.java (hunk 2 above)
Comment (jshook) [reply engine-clients/src/main/java/io/nosqlbench/engine/clients/grafana/analyzer/GrafanaRegionAnalyzer.java:46]: I'll remove it and push new changes.

**Review by jeffbanks — APPROVED** [review]
Changes to PR look good!

**Pull request merged**